KIP-500.md

```diff
@@ -0,0 +1,131 @@
+KIP-500 Early Access Release
+============================
+
+# Introduction
+It is now possible to run Apache Kafka without Apache ZooKeeper!  We call this mode [self-managed mode](https://cwiki.apache.org/confluence/display/KAFKA/KIP-500%3A+Replace+ZooKeeper+with+a+Self-Managed+Metadata+Quorum).  It is currently *EARLY ACCESS AND SHOULD NOT BE USED IN PRODUCTION*, but it is available for testing in the Kafka 2.8 release.
+
+When the Kafka cluster is in self-managed mode, it does not store its metadata in ZooKeeper.  In fact, you do not have to run ZooKeeper at all, because it stores its metadata in a Raft quorum of controller nodes.
+
+Self-managed mode has many benefits-- some obvious, and some not so obvious.  Clearly, it is nice to manage and configure one service rather than two services.  In addition, you can now run a single process Kafka cluster.  Most important of all, self-managed mode is more scalable.  We expect to be able to [support many more topics and partitions](https://www.confluent.io/kafka-summit-san-francisco-2019/kafka-needs-no-keeper/) in this mode.
+
+# Quickstart
+
+## Warning
+Self-managed mode in Kafka 2.8 is provided for testing only, *NOT* for production.  We do not yet support upgrading existing ZooKeeper-based Kafka clusters into this mode.  In fact, when Kafka 3.0 is released, it may not even be possible to upgrade your self-managed clusters from 2.8 to 3.0 without downtime.  There may be bugs, including serious ones.  You should *assume that your data could be lost at any time* if you try the early access release of KIP-500.
+
+## Generate a cluster ID
+The first step is to generate an ID for your new cluster, using the kafka-storage tool:
+
+~~~~
+$ ./bin/kafka-storage.sh random-uuid
+xtzWWN4bTjitpL3kfd9s5g
+~~~~
+
+## Format Storage Directories
+The next step is to format your storage directories.  If you are running in single-node mode, you can do this with one command:
+
+~~~~
+$ ./bin/kafka-storage.sh format -t xtzWWN4bTjitpL3kfd9s5g -c ./config/raft-combined.properties
```

Comment:
If this is "self-contained mode" it's a bit confusing for the unitiated that the example configs ares `raft-*`. Not everyone reading this README will know the role of Raft in this. Perhaps the file should be `self-contained-combined.properties`, though that's maybe too long.

Reply:
Perhaps it could be "nozk-combined.properties", etc. just to keep it simple? I think that should be a separate PR, though (do you want to file it)?

Reply:
How about `sm-{combined,broker,controller}.properties` where `sm` stands for `self-managed`?  I like that better than `nozk` becauae in a few years we will still have "zk" in the name; I think a positive name -- it's a config file for a self-managed case -- is better than a negative name (Look Mom, no ZooKeeper!)

Reply:
@rondagostino the problem I see with that is it only makes sense if someone's told you what sm means, `nozk` is more self-explanatory for anyone who knows the 40,000ft view that Kafka is moving away from ZooKeeper. Are we really going to support the Kafka 3.x series for _years_? In my ignorance, I would have expected quite strong pressure to move on to Kafka 4.0 (i.e. no ZK support) fairly quickly (maybe 12 months after 3.0?) simply to avoid having to maintain two implementations of so much code.

Do we even need a prefix at all though? We already have `server.properties`=ZK, adding new `broker.properties`, `controller.properties` and `combined.properties` (or `controller+broker.properties`) seems to me to be self-describing and both backwards and forwards compatible. The only slight issue would be a newbie accidentally running with `broker.properties` rather than `server.properties`, but that would fail safe without a quorum.

Reply:
> Do we even need a prefix at all though? We already have server.properties=ZK, adding new broker.properties, controller.properties and combined.properties (or controller+broker.properties) seems to me to be self-describing and both backwards and forwards compatible

I tend to agree with this suggestion.  I think using `{broker,controller,combined}.properties` for names would be a good way to go here.

Reply:
I agree that in the long term `{broker,controller,combined}.properties` are nice.  In the short term, though this could create confusion for people who want to run with ZK enabled (which is going to be most people, until we get out of "preview")  So let's go with nozk-* for now....